src/python/WMCore/Database/CMSCouch.py

```diff
@@ -346,6 +346,46 @@ def updateDocument(self, doc_id, design, update_func, fields={}, useBody=False):
                         (self.name, design, update_func, doc_id)
             return self.put(uri=updateUri, data=fields, decode=False)
 
+    def updateBulkDocuments(self, doc_ids, paramsToUpdate, updateLimits=1000):
+
+        uri = '/%s/_bulk_docs/' % self.name
+        conflictDocIDs = []
+        for ids in grouper(doc_ids, updateLimits):
+            # get original documens
+            docs = self.allDocs(options={"include_docs": True}, keys=ids)['rows']
+            data = {}
+            data['docs'] = []
+            for j in docs:
+                doc = {}
+                doc.update(j['doc'])
+                nestedDictUpdate(doc, paramsToUpdate)
+                data['docs'].append(doc)
+
+            if len(data['docs']) > 0:
+                retval = self.post(uri, data)
+                for result in retval:
+                    if result.get('error', None) == 'conflict':
+                        conflictDocIDs.append(result['id'])
```

Comment:
Seangchan, did you check whether the doc `id` is really reported back in case of a conflict?

Reply:
I haven't tested, I just check the document. (I am not sure how to create the conflict in test setting).

https://wiki.apache.org/couchdb/HTTP_Bulk_Document_API

One other option is we added option _all_or_nothing, but that will create the conflict.

Reply:
Don't we need the rev number for the doc update? If so, you could add the same doc id twice to the `data` to be updated. That might yield a conflict.

Yeah, that last line in this couch documentation is not very positive.

Reply:
rev number is not needed explicitly, The way update here woks is

1.  get all the documents (list of lds are given) first
2. rev is returned from first call and used for update.